### PR TITLE
Create watch command

### PR DESCRIPTION
**Pull request by ankitjena**
## Implementation
Use `chokidar` to watch for changes in `model` folder and run generate and db.

## Todo
Drop database implementation

### CODE REVIEW

File: packages/graphback-cli/src/helpers/generate.ts [hunk 1]
```diff
@@ -47,14 +47,16 @@ async function generateBackend(): Promise<void> {
...
 }
 
 /**
  * exported generate handler
  */
 export async function generate(): Promise<void> {
+  checkDirectory()
   await generateBackend()
   postCommandMessage()
+  process.exit(0)
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/generate.ts]: reason why this is here?

File: packages/graphback-cli/src/helpers/watch.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import * as execa from 'execa'
+import { Stats } from 'fs';
+import { checkDirectory } from './common';
+import { createResources } from './db';
+import { generateBackend } from './generate';
+
+const modelsPath = `${process.cwd()}/model`
+
+const watcher = chokidar.watch(modelsPath, {
+  ignored: /(^|[\/\\])\../,
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/watch.ts]: comment what we ignoring

File: packages/graphback-cli/src/helpers/init.ts [hunk 3]
```diff
@@ -114,4 +114,5 @@ Bootstraping graphql server :dizzy: :sparkles:`)
   await installDependencies(name)
   createDBConfig(name)
   logInfo(postSetupMessage(name))
+  process.exit(0)
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/init.ts]: Why this? This seems to be bad pattern for reusing this code

**Comment by wtrocki** [comment]
Executed watch but after chage it crashed with no error. Is chokidar an reputable package?

File: packages/graphback-cli/src/helpers/init.ts (hunk 3 above)
Comment (ankitjena) [reply packages/graphback-cli/src/helpers/init.ts]: After I added the watch command, all other commands got stuck after execution so added this as temporary fix for the workflow

**Comment by ankitjena** [comment]
> Executed watch but after chage it crashed with no error. Is chokidar an reputable package?

It's quite reputable actually

File: packages/graphback-cli/src/helpers/common.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { accessSync } from 'fs'
+import { logError } from '../utils'
+
+
+export const checkDirectory = (): void => {
+  try{
+    accessSync(`${process.cwd()}/model`)
+  } catch(err) {
+    logError(`model directory not found. Make you sure you are in the root of your project.`)
+    process.exit(0)
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/common.ts]: Please remove process exit.

File: packages/graphback-cli/src/helpers/db.ts [hunk 5]
```diff
@@ -39,6 +30,8 @@ export const createResources = async(): Promise<void> => {
 }
 
 export const createDB = async(): Promise<void> => {
+  checkDirectory()
   await createResources()
   logInfo("Database Created")
+  process.exit(0)
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/db.ts]: Remove this

File: packages/graphback-cli/src/helpers/watch.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  persistent: true
+});
+
+export const watch = async(): Promise<void> => {
+  checkDirectory()
+  execa('npm', ['run', 'start']).stdout.pipe(process.stdout);
+  watcher.on('change', async(path: string, stats: Stats) => {
+    await Promise.all([generateBackend(),createResources()])
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/watch.ts]: Create resources will override existing files? what this is doing?

File: packages/graphback-cli/src/helpers/watch.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+const modelsPath = `${process.cwd()}/model`
+
+const watcher = chokidar.watch(modelsPath, {
+  ignored: /(^|[\/\\])\../,
+  persistent: true
+});
+
+export const watch = async(): Promise<void> => {
+  checkDirectory()
+  execa('npm', ['run', 'start']).stdout.pipe(process.stdout);
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/watch.ts]: I think that we need to cancel return object.everytime this is the reason you got stuck initially. Lets do not make changes just to het things running but resolve problems in source

File: packages/graphback-cli/src/helpers/common.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { accessSync } from 'fs'
+import { logError } from '../utils'
+
+
+export const checkDirectory = (): void => {
+  try{
+    accessSync(`${process.cwd()}/model`)
+  } catch(err) {
+    logError(`model directory not found. Make you sure you are in the root of your project.`)
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/common.ts]: mkdir instead?

File: packages/graphback-cli/src/helpers/db.ts [hunk 9]
```diff
@@ -4,13 +4,26 @@ import { GraphQLBackendCreator, PostgresSchemaManager } from 'graphback';
...
-  const manager = new PostgresSchemaManager(dbConfig);
-  // tslint:disable-next-line: await-promise
-  await manager.getConnection().raw('DROP SCHEMA public CASCADE;')
-  // tslint:disable-next-line: await-promise
-  await manager.getConnection().raw('CREATE SCHEMA public;')
+  try {
+    const { dbConfig } = JSON.parse(readFileSync(`${process.cwd()}/config.json`, 'utf8'))
+    const manager = new PostgresSchemaManager(dbConfig);
+    // tslint:disable-next-line: await-promise
+    await manager.getConnection().raw('DROP SCHEMA public CASCADE;')
```
Comment (wtrocki) [inline packages/graphback-cli/src/helpers/db.ts]: 👍 This is what we need for the moment.
Problem with this is that this is postgres specific and public schema may not exist but let's not bother with that for the moment. This code will be replaced soon.
Some warning message here will be cool - your data is lost. 
Do not push that to production etc.

**Pull request merged**